src/hostapi/coreaudio_ios/pa_ios_core.c

```diff
@@ -0,0 +1,1429 @@
+ * EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF
+ * MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.
+ * IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR
+ * ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF
+ * CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION
+ * WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+ */
+
+/*
+ * The text above constitutes the entire PortAudio license; however,
+ * the PortAudio community also makes the following non-binding requests:
+ *
+ * Any person wishing to distribute modifications to the Software is
+ * requested to send the modifications to the original developer so that
+ * they can be incorporated into the canonical version. It is also
+ * requested that these non-binding requests be included along with the
+ * license above.
+ */
+
+/**
+ @file pa_ios_core
+ @ingroup hostapi_src
+ @author Bjorn Roche
+ @brief AUHAL implementation of PortAudio
+*/
+
+#include "pa_ios_core_internal.h"
+
+#include <string.h>
+
+#include <libkern/OSAtomic.h>
+
+#include <mach/mach_time.h>
+
+#include "pa_ios_core.h"
+#include "pa_ios_core_utilities.h"
+#include "pa_ios_core_blocking.h"
+
+/* prototypes for functions declared in this file */
+PaError PaIosCore_Initialize(PaUtilHostApiRepresentation ** hostApi, PaHostApiIndex index);
+
+static void Terminate(struct PaUtilHostApiRepresentation *hostApi);
+static PaError IsFormatSupported(struct PaUtilHostApiRepresentation *hostApi,
+                                 const PaStreamParameters * inputParameters,
+                                 const PaStreamParameters * outputParameters,
+                                 double sampleRate);
+static PaError OpenStream(struct PaUtilHostApiRepresentation *hostApi,
+                          PaStream ** s,
+                          const PaStreamParameters * inputParameters,
+                          const PaStreamParameters * outputParameters,
+                          double sampleRate,
+                          unsigned long framesPerBuffer,
+                          PaStreamFlags streamFlags,
+                          PaStreamCallback * streamCallback,
+                          void *userData);
+static PaError CloseStream(PaStream * stream);
+static PaError StartStream(PaStream * stream);
+static PaError StopStream(PaStream * stream);
+static PaError AbortStream(PaStream * stream);
+static PaError IsStreamStopped(PaStream * s);
+static PaError IsStreamActive(PaStream * stream);
+static PaTime GetStreamTime(PaStream * stream);
+static OSStatus AudioIOProc(void *inRefCon,
+    AudioUnitRenderActionFlags * ioActionFlags,
+    const AudioTimeStamp * inTimeStamp,
+    UInt32 inBusNumber,
+    UInt32 inNumberFrames,
+    AudioBufferList * ioData);
+static double GetStreamCpuLoad(PaStream * stream);
+
+/*
+ * Callback called when starting or stopping a stream.
+ */
+static void
```

Comment:
Typically we put the qualifier and return type on the same line as the function name, could you do that please?